apps/hash-frontend/src/pages/notes.page/editable-quick-note.tsx

```diff
@@ -0,0 +1,372 @@
+    ] as TextToken[];
+
+    return textTokens.reduce(
+      (prevText, current) =>
+        current.tokenType === "text" ? `${prevText}${current.text}` : prevText,
+      "",
+    );
+  }
+
+  return "";
+};
+
+export const EditableQuickNote: FunctionComponent<{
+  displayLabel?: boolean;
+  displayActionButtons?: boolean;
+  quickNoteEntityWithCreatedAt: QuickNoteEntityWithCreatedAt;
+  quickNoteSubgraph?: Subgraph<EntityRootType>;
+  refetchQuickNotes?: () => Promise<void>;
+  autofocus?: boolean;
+}> = ({
+  displayLabel = true,
+  displayActionButtons = true,
+  quickNoteEntityWithCreatedAt,
+  quickNoteSubgraph,
+  refetchQuickNotes,
+  autofocus = false,
+}) => {
+  const { authenticatedUser } = useAuthenticatedUser();
+
+  const [archiveEntity] = useMutation<
+    ArchiveEntityMutation,
+    ArchiveEntityMutationVariables
+  >(archiveEntityMutation);
+
+  const [convertedPage, setConvertedPage] = useState<PageWithParentLink>();
+  const [isConvertingPage, setIsConvertingPage] = useState(false);
+
+  const { updateEntity } = useBlockProtocolUpdateEntity();
+
+  const [isConvertToPageModalOpen, setIsConvertToPageModalOpen] =
+    useState(false);
+
+  const apolloClient = useApolloClient();
+
+  const refetchPageTree = useCallback(async () => {
+    await apolloClient.refetchQueries({ include: [getAccountPagesTree] });
+  }, [apolloClient]);
+
+  const { quickNoteEntity } = quickNoteEntityWithCreatedAt;
+
+  const blockCollectionEntityId = quickNoteEntity.metadata.recordId.entityId;
+
+  const { data } = useQuery<GetEntityQuery, GetEntityQueryVariables>(
+    getEntityQuery,
+    {
+      variables: {
+        entityId: blockCollectionEntityId,
+        ...zeroedGraphResolveDepths,
+        hasLeftEntity: { incoming: 2, outgoing: 2 },
+        hasRightEntity: { incoming: 2, outgoing: 2 },
+      },
+      fetchPolicy: "cache-and-network",
+    },
+  );
+
+  const mostRecentContents = useMemo(
+    () =>
+      data?.getEntity
+        ? getBlockCollectionContents({
+            blockCollectionEntityId,
+            blockCollectionSubgraph: data.getEntity
+              .subgraph as Subgraph<EntityRootType>,
+          })
+        : undefined,
+    [blockCollectionEntityId, data],
+  );
+
+  const contents = useMemo(
+    () =>
+      mostRecentContents ??
+      (quickNoteSubgraph
+        ? getBlockCollectionContents({
+            blockCollectionEntityId,
+            blockCollectionSubgraph: quickNoteSubgraph,
+          })
+        : undefined),
+    [blockCollectionEntityId, quickNoteSubgraph, mostRecentContents],
+  );
+
+  const numberOfBlocks = useMemo(
+    () => (mostRecentContents ? mostRecentContents.length : undefined),
+    [mostRecentContents],
+  );
+
+  const textBlocks = useMemo(
+    () =>
+      mostRecentContents?.filter(({ rightEntity }) =>
+        isHashTextBlock(rightEntity.componentId),
+      ),
+    [mostRecentContents],
+  );
+
+  const text = useMemo(
```

Comment:
This seems to undercount words.. maybe something to do with the general bugginess of save (not introduced by this PR), since it appears to be counting only the words that get persisted. Possibly blocks are being saved before all words in them are typed and then not updated again

https://github.com/hashintel/hash/assets/37743469/b5d303e5-285e-4a83-bb8e-c39c6232f8c9

Reply:
You're right that the word count will only be updated once the update has persisted, this is because the parent component relies on a GQL refetch query call to be made by the `save.ts` function to get the updated page contents.

Without a better alternative, I'm not sure how I can get the word count to reflect what the user is currently typing. The calculation of how many words itself though should be robust, the reason the number of words decreases when you clicked "Create" in your recording is because the most recent update did not get persisted (another issue I flagged in the description, as I can't disable the form from submitting before the last save has completed).